fritz_interview.py

```diff
@@ -0,0 +1,70 @@
+# Lucky Dollar Store Coding Challenge
+import random
+names = ["Eric", "Chris", "Dan", "Jameson"]
+		
+
+def record_customer_purchase(customer):
+     # Fill this in
+     print("- Added new raffle ticket for \"%s\"" % customer)
+     return customer
+
+
+def can_win(customer, pool):
+	if not pool:  
+		return True
+	elif customer in pool.keys():
+		if pool[customer] == 0:
+			return True
+	elif customer not in pool.keys():
+		return True
+	return False
+
+
+def run_weekly_raffle(num_winners, num_purchases):
+	# pick winner(s)
+	pool = []
+	winning_customer = []
+	for n in range(0, num_purchases):
```

Comment:
Let's say that **record_customer_purchase** and **run_weekly_raffle** are the methods that the store manager would like to use. In that case, we don't want to generate random purchases in this method.

Try separating randomly generated test data (purchases) from the business logic. Say that the user wants to use these methods like the following (roughly):

```
# Add raffle tickets
record_customer_purchase("Hally")
record_customer_purchase("Jerry")
record_customer_purchase("Bill")

# Pick one winner
run_weekly_raffle(1)
```

What would be a good way to handle that? Also it doesn't have to be the exact same syntax or method definition. You can use those methods in a class however you'd like to structure it. The primary goal is to be able to record a customer purchase and then choose winner(s) from that raffle pool.

Reply:
I am thinking I am supposed to encapsulate the raffle in a class in order to avoid using a global variable for the raffle pool. Would you rather it be in a class or can I just use global functions and variables?

I am setting to work on the Raffle class now.

Reply:
Let's try creating a class.

For testing this out, you can use the main function and keep number of purchases, customer names, etc separated from the raffle methods.

```
def main():
     Raffle raffle = Raffle()
      # Record 3 purchases
     raffle.record_customer_purchase("Hally")
     raffle.record_customer_purchase("Jerry")
     raffle.record_customer_purchase("Bill")

     # Pick one winner
    raffle.run_weekly_raffle(1)

    # ....
```

Reply:
Great! Would you want a Raffle class that can be initialized with a number of drawings? For instance 
`raffle = Raffle(6)` 
would do 6 raffle drawings in the record_customer_purchase method. 

Alternatively, I could remove the outer loop from my original code 
`for i in range(1, num_drawings + 1):`
and there would be 1 drawing per call of record_customer_purchase, leaving the number of drawings up to the store manager. The number of drawings would be a property of the Raffle class that would increase by 1 with each call of record_customer_purchase.

I have already removed the randomly generated num_purchases and global list of names.